tools/introspection/source/introspection_app.cpp

```diff
@@ -102,15 +102,16 @@ void IntrospectionApp::parseCmdLineArguments(int argc,
         case 't':
         {
             uint64_t newUpdatePeriodMs;
-            if (convert::fromString(optarg, newUpdatePeriodMs))
-            {
-                iox::units::Duration rate = iox::units::Duration::fromMilliseconds(newUpdatePeriodMs);
-                updatePeriodMs = bounded(rate, MIN_UPDATE_PERIOD, MAX_UPDATE_PERIOD);
-            }
-            else
+            auto result = convert::from_string<uint64_t>(optarg);
+            if (!result.has_value())
             {
                 std::cout << "Invalid argument for 't'! Will be ignored!";
+                break;
             }
+
+            newUpdatePeriodMs = result.value();
```

Comment:
This can also become `const auto ...` ... or the whole part could be replaced by `and_then/or_else` ... up to you

Reply:
See [809dddc](https://github.com/eclipse-iceoryx/iceoryx/pull/2135/commits/809dddca229ad98fa46ad08ccfd271927ae83cab).